hack/ci/setup-kind-cluster.sh

```diff
@@ -0,0 +1,232 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The Machine Controller Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+source hack/lib.sh
+
+echodate "Setting up kind cluster..."
+
+if [ -z "${JOB_NAME:-}" ] || [ -z "${PROW_JOB_ID:-}" ]; then
+  echodate "This script should only be running in a CI environment."
+  exit 1
+fi
+
+export KIND_CLUSTER_NAME="${KIND_CLUSTER_NAME:-machine-controller}"
+
+start_docker_daemon_ci
+
+# Prevent mtu-related timeouts
+echodate "Setting iptables rule to clamp mss to path mtu"
+iptables -t mangle -A POSTROUTING -p tcp --tcp-flags SYN,RST SYN -j TCPMSS --clamp-mss-to-pmtu
+
+# Make debugging a bit better
+echodate "Configuring bash"
+cat << EOF >> ~/.bashrc
+# Gets set to the CI clusters kubeconfig from a preset
+unset KUBECONFIG
+
+cn() {
+  kubectl config set-context --current --namespace=\$1
+}
+
+kubeconfig() {
+  TMP_KUBECONFIG=\$(mktemp);
+  kubectl get secret admin-kubeconfig -o go-template='{{ index .data "kubeconfig" }}' | base64 -d > \$TMP_KUBECONFIG;
+  export KUBECONFIG=\$TMP_KUBECONFIG;
+  cn kube-system
+}
+
+# this alias makes it so that watch can be used with other aliases, like "watch k get pods"
+alias watch='watch '
+alias k=kubectl
+alias ll='ls -lh --file-type --group-directories-first'
+alias lll='ls -lahF --group-directories-first'
+source <(k completion bash )
+source <(k completion bash | sed s/kubectl/k/g)
+EOF
+
+# Find external IP of node where this pod is running
+echodate "Retrieving the external node IP where this pod is scheduled"
+export NODE_NAME=$(kubectl --kubeconfig /etc/kubeconfig/kubeconfig get pods -l prow.k8s.io/id=$PROW_JOB_ID -o jsonpath="{.items..spec.nodeName}")
+export NODE_IP=$(kubectl --kubeconfig /etc/kubeconfig/kubeconfig get node $NODE_NAME -o jsonpath="{.status.addresses[?(@.type=='ExternalIP')].address}")
+
+if [ -z "$NODE_NAME" ] || [ -z "$NODE_IP" ]; then
+  echodate "This script was unable to determine the external IP for kube-apiserver."
+  exit 1
+fi
+
+# Create kind cluster
+TEST_NAME="Create kind cluster"
+
+echodate "Preloading the kindest/node image"
+docker load --input /kindest.tar
+
+echodate "Creating the kind cluster"
+export KUBECONFIG=~/.kube/config
+
+beforeKindCreate=$(nowms)
+
+cat << EOF > kind-config.yaml
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+name: "${KIND_CLUSTER_NAME}"
+networking:
+  apiServerAddress: "0.0.0.0"
+  disableDefaultCNI: true # disable kindnet
```

Comment:
`i++ // increment i by 1`

_why_ do we disable kindnet?

Reply:
We disable it in favor of flannel.

Reply:
Why do we do that? What do we get with Flannel that's not available with Kindnet?